pkg/hubble/helpers_test.go

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package hubble
+
+import (
+	"fmt"
+	"testing"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	"github.com/cilium/cilium/test/helpers"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseFlowFilters(t *testing.T) {
+	testCases := []struct {
+		name    string
+		arg     string
+		want    []*flowpb.FlowFilter
+		wantErr bool
+	}{
+		// valid
+		{name: "empty"},
+		{name: "whitespaces", arg: "   "},
+		{name: "one json", arg: "{}", want: []*flowpb.FlowFilter{{}}},
+		{name: "one json whitespaces", arg: "{}  ", want: []*flowpb.FlowFilter{{}}},
+		{name: "two jsons", arg: "{}{}", want: []*flowpb.FlowFilter{{}, {}}},
+		{name: "two jsons whitespaces", arg: "{}  {}", want: []*flowpb.FlowFilter{{}, {}}},
+		{name: "two jsons whitespaces", arg: "{}  {}   ", want: []*flowpb.FlowFilter{{}, {}}},
```

Comment:
Is this what we want? I suspect we may need to allow whitespace as a delimiter due to backwards compatibility, but this also means our JSON filters cannot have any whitespace in them.

Eg the space after `:` means this won't work, even though it's valid JSON:
```
{"is_reply": [true]}
```

Reply:
Edit: i think `json.Decoder` actually handles both, which I didn't realize. In any case, let's add tests for objects separated by newlines and include some objects with spaces in them.

Reply:
Yeah good call, I added more test cases with whitespaces inside/outside json objects.